sql/src/main/java/io/crate/metadata/settings/CrateSettings.java

```diff
@@ -287,11 +288,46 @@ Settings settings() {
     private Map<String, ReferenceImplementation> buildReferenceTree() {
         Map<String, ReferenceImplementation> referenceMap = new HashMap<>(BUILT_IN_SETTINGS.size());
         for (CrateSetting crateSetting : BUILT_IN_SETTINGS) {
+            if (crateSetting.isGroupSetting()) {
+                Map<String, Settings> settingsMap = initialSettings.getGroups(crateSetting.getKey(), true);
+                for (Map.Entry<String, Settings> entry : settingsMap.entrySet()) {
+                    buildGroupSettingReferenceTree(crateSetting.getKey(), entry, referenceMap);
+                }
+            }
             buildReferenceTree(referenceMap, crateSetting);
         }
         return referenceMap;
     }
 
+    @VisibleForTesting
+    void buildGroupSettingReferenceTree(String prefix, Map.Entry<String, Settings> settingEntry, Map<String, ReferenceImplementation> referenceMap) {
+        Map<String, String> settingsMap = settingEntry.getValue().getAsMap();
+
+        //this is a nested setting
+        if (!settingsMap.isEmpty()) {
+            //we need to build the reference tree for the current setting
+            buildReferenceTree(referenceMap,
+                CrateSetting.of(Setting.groupSetting(prefix + settingEntry.getKey() + ".",
+                    Setting.Property.NodeScope),
+                    DataTypes.OBJECT));
+            //build the reference tree for every child setting
+            for (Map.Entry<String, String> entry : settingsMap.entrySet()) {
+                String nestedPrefix = prefix + settingEntry.getKey() + "." + entry.getKey();
+
+                buildReferenceTree(referenceMap,
+                    CrateSetting.of(Setting.simpleString(nestedPrefix,
+                        Setting.Property.NodeScope),
+                        DataTypes.STRING));
+            }
+        } else {
+            //this is a simple setting 
+            buildReferenceTree(referenceMap,
```

Comment:
what is this case for? this is not needed and afaik this would result in that unsupported settings can be registered.